.github/workflows/backport-command.yml

```diff
@@ -64,11 +110,196 @@ jobs:
+        env:
+          REVIEWERS: ${{ toJson(github.event.client_payload.pull_request.requested_reviewers) }}
+        id: reviewers
+        run: echo ::set-output name=reviewers::$(echo "$REVIEWERS" | jq -r '.[].login' | paste -s -d ',' -)
+
+      - name: Get commits of PR
+        if: needs.backport-type.outputs.commented_on == 'pr'
+        env:
+          BACKPORT_PR_NUMBER: ${{ github.event.client_payload.pull_request.number }}
+          GITHUB_TOKEN: ${{ secrets.ACTIONS_BOT_TOKEN }}
+        id: backport_commits
+        run: |
+          backport_commits=$(gh api "repos/$TARGET_ORG/$TARGET_REPO/pulls/$BACKPORT_PR_NUMBER/commits" --jq .[].sha | paste -s -d ' ' -)
+          echo ::set-output name=backport_commits::$backport_commits
+
+      - uses: actions/checkout@v3
+        if: needs.backport-type.outputs.commented_on == 'pr'
+        with:
+          repository: ${{ steps.user.outputs.username }}/${{ steps.user.outputs.repo }}
+          token: ${{ secrets.ACTIONS_BOT_TOKEN }}
+
+      - name: Backport commits and get details
+        if: needs.backport-type.outputs.commented_on == 'pr'
+        env:
+          PR_NUMBER: ${{ github.event.client_payload.pull_request.number }}
+          GITHUB_TOKEN: ${{ secrets.ACTIONS_BOT_TOKEN }}
+          BACKPORT_BRANCH: ${{ steps.branch.outputs.branch }}
+          ORIG_TITLE: ${{ github.event.client_payload.github.payload.issue.title }}
+          BACKPORT_COMMITS: ${{ steps.backport_commits.outputs.backport_commits }}
+          IS_MERGED: ${{ github.event.client_payload.pull_request.merged }}
+          PR_BASE_BRANCH: ${{ github.event.client_payload.pull_request.base.ref }}
+          REPO_DEFAULT_BRANCH: ${{ github.event.client_payload.pull_request.base.repo.default_branch }}
+        id: pr_details
+        run: |
+          if [[ "$IS_MERGED" != true ]]; then
+            msg="The pull request is not merged yet. Cancelling backport..."
+            echo "$msg"
+            gh pr comment "$PR_NUMBER" \
+              --body "$msg" \
+              --repo "$TARGET_ORG/$TARGET_REPO"
+            exit 1
+          elif [[ "$PR_BASE_BRANCH" != "$REPO_DEFAULT_BRANCH" ]]; then
+            msg="The pull request's base branch is not the default one. Cancelling backport..."
+            echo "$msg"
+            gh pr comment "$PR_NUMBER" \
+              --body "$msg" \
+              --repo "$TARGET_ORG/$TARGET_REPO"
+            exit 1
+          fi
+          fixing_issue_urls=$(gh api graphql -f query='{
+              resource(url: "https://github.com/'${TARGET_ORG}'/'${TARGET_REPO}'/pull/'${PR_NUMBER}'") {
+                ... on PullRequest {
+                  closingIssuesReferences(first: 20) {
+                    nodes {
+                      url
+                    }
+                  }
+                }
+              }
+            }' --jq .data.resource.closingIssuesReferences.nodes.[].url)
+          # ensure unique branch
+          suffix=$(echo $(($RANDOM % 1000)))
```

Comment:
i feel this is prone to collision especially if we don't delete branches after PR merges. would be safer to use timestamp from `date '+%s'`. can be addressed in follow-up PR.